agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/statsbeat/StatsbeatModule.java

```diff
@@ -46,51 +47,81 @@
   private final NetworkStatsbeat networkStatsbeat;
   private final AttachStatsbeat attachStatsbeat;
   private final FeatureStatsbeat featureStatsbeat;
+  private final FeatureStatsbeat instrumentationStatsbeat;
 
   private final AtomicBoolean started = new AtomicBoolean();
+  private volatile boolean disabledAll;
+
+  private volatile TelemetryClient telemetryClient;
 
   private StatsbeatModule() {
     customDimensions = new CustomDimensions();
     networkStatsbeat = new NetworkStatsbeat(customDimensions);
     attachStatsbeat = new AttachStatsbeat(customDimensions);
-    featureStatsbeat = new FeatureStatsbeat(customDimensions);
+    featureStatsbeat = new FeatureStatsbeat(customDimensions, FeatureType.FEATURE);
+    instrumentationStatsbeat = new FeatureStatsbeat(customDimensions, FeatureType.INSTRUMENTATION);
   }
 
   public void start(TelemetryClient telemetryClient, Configuration config) {
     if (started.getAndSet(true)) {
       throw new IllegalStateException("initialize already called");
     }
 
-    long intervalSeconds = config.internal.statsbeat.intervalSeconds;
-    long featureIntervalSeconds = config.internal.statsbeat.featureIntervalSeconds;
+    if (config.internal.statsbeat.disabledAll) {
+      // disabledAll is an internal emergency kill-switch to turn off Statsbeat completely when
+      // something goes wrong.
+      // this happens rarely.
+      disabledAll = true;
+      return;
+    }
+
+    if (this.telemetryClient == null) {
+      this.telemetryClient = telemetryClient;
+      networkStatsbeat.initInstrumentationKeyCounterMap(
+          Arrays.asList(telemetryClient.getInstrumentationKey()));
+    }
+
+    long shortIntervalSeconds = config.internal.statsbeat.shortIntervalSeconds;
+    long longIntervalSeconds = config.internal.statsbeat.longIntervalSeconds;
 
     scheduledExecutor.scheduleWithFixedDelay(
         new StatsbeatSender(networkStatsbeat, telemetryClient),
-        intervalSeconds,
-        intervalSeconds,
+        shortIntervalSeconds,
+        shortIntervalSeconds,
         TimeUnit.SECONDS);
     scheduledExecutor.scheduleWithFixedDelay(
         new StatsbeatSender(attachStatsbeat, telemetryClient),
-        intervalSeconds,
-        intervalSeconds,
+        60,
+        longIntervalSeconds,
```

Comment:
it is intentional this is the only one with the short initial delay?

Reply:
yes, only attach we need to do this and the azure metadata service.